programs/utilities/uuids.scl

```diff
@@ -0,0 +1,57 @@
+// Returns a new UUID int array
+generate_uuid() -> (
+    uuid = map(range(4), floor(rand(2^32)));
+    uuid:1 = bitwise_and(uuid:1, 0xffff0fff); // clear version
+    uuid:1 = bitwise_or(uuid:1, 0x00004000);  // set to version 4
+    uuid:2 = bitwise_and(uuid:2, 0x3fffffff); // clear variant
+    uuid:2 = bitwise_or(uuid:2, 0x80000000);  // set to IETF variant
+    uuid
+);
+
+// Returns a new UUID string
+generate_uuid_string() -> uuid_to_string(generate_uuid());
+
+// Converts UUID int array to UUID string
+uuid_to_string(uuid) -> (
+    if( type(uuid) != 'list' ||
+        length(uuid) != 4 ||
+        reduce(uuid, _a || type(_) != 'number', false),
+    // then
+        exit('Invalid UUID: ' + uuid)
```

Comment:
Same here, this being largely cos that way if it's being used within the program then it won't explode if the programmer sends a wrong input, especially given that it's a ibrary